### ISSUE TITLE
Multitasking Loss

### ISSUE THREAD

**Issue by shrimonmuke0202** [UKPLab/sentence-transformers#2509]
Hello,

I would like to utilize InfoNCE loss in conjunction with cross-entropy loss. How can this be achieved using the Sentence Transformer library? Essentially, my goal is to have the final loss calculated as: Final Loss = InfoNCE Loss + Cross-Entropy Loss.

**Comment by shrimonmuke0202** [comment]
Thank you. I'm planning to implement a multitask loss function, L, which is a combination of the InfoNCE Loss and the Cross-Entropy Loss, weighted by the factors alpha and beta, respectively. To achieve this, I am using two distinct text encoders: one computes the InfoNCE Loss, and the other computes the Cross-Entropy Loss.

**Comment by tomaarsen** [comment]
I see. The InfoNCE should be equivalent to [MultipleNegativesRankingLoss](https://www.sbert.net/docs/package_reference/losses.html#multiplenegativesrankingloss), and I think Cross-Entropy is fairly unusual for embedding models, as it's not a contrastive loss. The [SoftmaxLoss](https://www.sbert.net/docs/package_reference/losses.html#softmaxloss) is probably closest: it uses a linear layer to reduce the embeddings into a much smaller size, and then uses CrossEntropyLoss to compare them to integer (class) labels. It is generally considered to not be a very strong loss function.

- Tom Aarsen
